File: palserver/internal/room/handler.go
package room

import (
	"github.com/zhangga/chatpal/palserver/internal/msg"
)

type Handler func(c *conn, msgType int, content []byte) (msg.Message, error)

var msgHandlers = map[msg.Id]Handler{
	msg.Login:      handleLogin,
	msg.CreateRoom: handleCreateRoom,
	msg.ListRoom:   handleListRoom,
}

File: palmini/miniprogram/services/message/handler.ts
import { Message, MessageId } from "./message";

// 定义消息处理函数的类型
type MessageHandler = (message: Message) => void;

interface IMessageHandler {
  [id: number]: MessageHandler
}

const handleLogin: MessageHandler = (message: Message) => {
  console.log('======= 处理登录消息返回: ', message);
};

const handleError: MessageHandler = (message: Message) => {
  console.error('======= 错误处理消息: ', message);
};

// 消息处理
export const MessageHandlers: IMessageHandler = {
  [MessageId.Login]: handleLogin,
}
